Smsalert

pamzey opened the pull request:
I have added sms controller and service to be used for notifications

pamzey committed:
created immunization report engine

pamzey committed:
added follow controller

pamzey committed:
fixed the search soundex

pamzey committed:
soundex fix

pamzey committed:
Merge branch 'chore/find_patients_pagination' of https://github.com/HISMalawi/BHT-EMR-API into followups

pamzey committed:
Merge branch 'chore/find_patients_pagination' of https://github.com/HISMalawi/BHT-EMR-API into followups

pamzey committed:
added sms alert for the next apointment date

pamzey committed:
Merge branch 'chore/find_patients_pagination' of https://github.com/HISMalawi/BHT-EMR-API into smsalert

fchiyenda commented on config/initializers/sidekiq.rb:
```diff
@@ -0,0 +1,12 @@
+# config/initializers/sidekiq.rb
+Sidekiq.default_worker_options = { 'retry' => true } # Retry indefinitely
+
```
Line crushing when trying to run the application
```suggestion
Sidekiq.default_job_options = { 'retry' => true } # Retry indefinitely
```

pamzey replied:
This has been resolved

fchiyenda commented on app/controllers/api/v1/immunization_reports_controller.rb:
```diff
@@ -0,0 +1,47 @@
+      private
+
+      def validate_params
+        permitted = params.permit(%i[start_date end_date name quarter year]).to_h
```
I think we should also handle required params

pamzey replied:
This was handled

pamzey replied:
I have removed this file

fchiyenda commented on app/services/immunization_service/send_sms_service.rb:
```diff
@@ -0,0 +1,43 @@
+                "kut muzalandileso katemela.\n" \
+                "Zikomo."
+
+      uri = URI.parse("https://gateway.seven.io/api/sms")
```
I am thinking API key and gateway should be in a configuration file. eg application.yml

pamzey replied:
I have added all the suggested changes tried it and the job logs seems to OK

pamzey committed:
added a job to the send sms

fchiyenda commented on config/initializers/load_application_yml.rb:
```diff
@@ -0,0 +1,5 @@
+# config/initializers/load_application_yml.rb
+config = YAML.load_file(Rails.root.join('config', 'application.yml'))[Rails.env]
```
Let us ensure YAML aliases are enabled usually it breaks the code:
```suggestion
config = YAML.safe_load(File.read(Rails.root.join('config', 'application.yml')), aliases: true)[Rails.env]
```

pamzey replied:
This was handled

fchiyenda commented on config/application.yml.example:
```diff
@@ -1,104 +0,0 @@
-# Comment out parameter below to enable syncing with lims through the REST API.
```
Chief I think you accidentally deleted application.yml.example

pamzey replied:
i have replaced the file and added the suggestion

pamzey replied:
i have added the application.yml.example file

pamzey committed:
added application.yml file

fchiyenda commented on app/jobs/send_sms_job.rb:
```diff
@@ -0,0 +1,40 @@
+        http.request(request)
+      end
+
+      Rails.logger.info("SMS Gateway Response........pamzey: #{response.code} - #{response.body}")
```
Let us remove the personalization

pamzey replied:
i have removed the error logger

fchiyenda replied:
Meant removing your name in the code

pamzey replied:
This has been fixed

pamzey committed:
removed error handling log

pamzey committed:
added sms_gateway in the configuration file

fchiyenda reviewed (changes requested):
Let us not push the key to github

pamzey committed:
added details for getting sms api key

pamzey commented:
I have added guide to the read me how to get the sms api key

fchiyenda reviewed (commented):
> i have added the application.yml.example file 

I think you did not replace the first line which is informative for people who are setting up.

pamzey committed:
removed sms gateway provider

pamzey committed:
added line to guide user where to get the sms key

pamzey commented:
I have removed the line. The reason I left it there was to inform the user about where we purchased our SMS credit, in case they are new. However, I understand the security issue, so I have removed it.

fchiyenda commented on README.md:
```diff
@@ -407,3 +407,10 @@ See the following for more:
+
+# SMS Gateway for the immunization app
+  * You need to ask any of the team members for the api key
+  * To get the api key from https://gateway.seven.io
```
This not clear please explain in detail how one can get the api key and also the url for the gateway

pamzey replied:
I have revised the guide for getting the sms tell me if it makes sense

pamzey committed:
added read me for sms gateway

pamzey committed:
Merge branch 'chore/find_patients_pagination' of https://github.com/HISMalawi/BHT-EMR-API into smsalert

pamzey committed:
removed file immunization_reports_controller.rb
